sympy/logic/boolalg.py

```diff
@@ -1004,6 +1016,18 @@ def to_nnf(self, simplify=True):
                 args.append(Or(*clause))
         return And._to_nnf(*args, simplify=simplify)
 
+    def _rewrite_as_And(self):
+        arguments = list(ordered(self._argset))
+        if len(arguments) == 2:
+            return Or(And(Not(arguments[0]), arguments[1]), And(Not(arguments[1]), arguments[0]))
+        else:
+            for i in range(len(arguments)-1):
+                eq = Or(And(Not(arguments[i]), arguments[i+1]), And(Not(arguments[i+1]), arguments[i])).simplify()
```

Comment:
Is Xor defined for more than two variables? In that case, how? (I would assume parity, although it can be argued if there actually is an XOR-function for more than two variables.

Reply:
I searched regarding XOR of more than 2 inputs then it was showing that it exists. So I did this.  Without simplify() output was not in simplified form. We can do one thing rather using simplify multiple times we can use it while returning.

Reply:
I checked the documentation and Xor seems to be parity/add-modulo-2 so OK with multiple inputs (although the doubt was more from a formal perspective, it makes senses from a practical perspective).

OK. I guess that one can figure out a way to write it by looking at e.g. SOPform and the line 
`Or(*[_convert_to_varsSOP(x, variables) for x in essential])` where essential would be a list of all combinations of len(args) variables with an odd number of ones (can be generated by ibin, see sympy.utilities.iterables, and some filtering) and variables are args.

Not sure if it is required though. But feel free to give it a try. For a three-input Xor, the list `essential` should look like `[[1, 0, 0], [0, 1, 0], [0, 0, 1], [1, 1, 1]]` and so on.

Reply:
Can we reuse one of the other methods here, like to_cnf or to_dnf?

Reply:
It does not give the same result if the arguments are boolean functions in themselves. 

`to_cnf(Xor(And(x, z), y, z))`
![image](https://user-images.githubusercontent.com/8114497/60043425-54acaa80-96c0-11e9-8856-30fb929dfad3.png)

`Xor(And(x, z), y, z).rewrite(And)`
![image](https://user-images.githubusercontent.com/8114497/60043459-68f0a780-96c0-11e9-87d2-fb273f4fd4c3.png)

(With reservations for that I have messed up my installation running two different sources...)

They both simplify to the same expression though. It think the rewrite approach is slightly more suitable here as one most likely would except the expression on a certain form (not cnf, but as a rewrite based on the original expression). (I'm also a bit surprised that the z and ~z is simplified in the last cnf-expression.)

Of course one can argue which is the "best" form... One way is of course to just recreate a minimal cnf/nnf (SOP/POS) expression, but since there are other methods for that, I think this makes sense.